baseplate/lib/metrics.py

```diff
@@ -512,3 +560,40 @@ def metrics_client_from_config(raw_config: config.RawConfig) -> Client:
 
     # pylint: disable=maybe-no-member
     return make_client(cfg.metrics.namespace, cfg.metrics.endpoint)
+
+
+def metrics_observer_from_config(raw_config: config.RawConfig, client: Client) -> Any:
```

Comment:
1. The metrics library doesn't generally know the observers even exist, so it feels a bit odd to have this in this file.
2. Since we're already doing branching like this in `configure_observers`, I think we can probably skip the indirection here? Then we just ask the right observer directly to configure itself. A common pattern in baseplate for this is a `.from_config()` classmethod:

```python
class TaggedMetricsBaseplateObserver(BaseplateObserver):
    @classmethod
    def from_config_and_client(cls, raw_config, client):
        cfg = config.parse_config(raw_config, {
            "metrics_observer:" {
               ...
            }
        })
        return cls(cfg, client)

    def __init__(self, cfg, client):
        ...
```

Reply:
configure_observers() will decide which metrics observer to use

Reply:
will require configure_observers use, logic for which metrics observer to attach will occur there

Reply:
remove metrics_observer_from_config as parsing of cfg will occur inside TaggedMetricsObserver